ros2bag/ros2bag/verb/burst.py

```diff
@@ -60,7 +60,7 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             help='Path to a yaml file defining storage specific configurations. '
                  'See storage plugin documentation for the format of this file.')
         parser.add_argument(
-            '--start-offset', type=check_positive_float, default=0.0,
+            '--start-offset', type=check_not_negative_float, default=0.0,
```

Comment:
Do we actually needs this to include `0`? that case we just do not need to specify.

Reply:
It makes sense for launchfiles which also have a default argument for example.

Reply:
@Timple thanks for the info. can you share the actual example?

Reply:
Example:

```xml
<launch>
    <arg name="bag_file"/>

    <arg name="clock" default="100" />
    <arg name="rate" default="1.0" />
    <arg name="start_offset" default="0.0" />

    <set_parameter name="use_sim_time" value="true" />

    <executable name="rosbag" cmd="ros2 bag play --clock $(var clock) --rate $(var rate) --start-offset $(var start_offset) $(var bag_file)" />
</launch>
```

For now, we work around this by setting the default of the `start_offset` to a small value. But since the default (not specified) is zero, I think it should also accept this value as an input.

Reply:
thanks! i am not sure the constraints of launch description, so you saying it must be set, right?

> But since the default (not specified) is zero, I think it should also accept this value as an input.

yes, this is true. default should be accepted if specified...

Reply:
A default is not necessary but this would enforce the user to always pass the `start_offset` to the launch file which is not desired.